src/map/map.cpp

```diff
@@ -1736,8 +1736,8 @@ int map_search_freecell(struct block_list *src, int16 m, int16 *x,int16 *y, int1
 	}
 
 	while(tries--) {
-		*x = (rx >= 0)?(rnd()%rx2-rx+bx):(rnd()%(mapdata->xs-2)+1);
-		*y = (ry >= 0)?(rnd()%ry2-ry+by):(rnd()%(mapdata->ys-2)+1);
+		*x = (rx >= 0) ? (rnd(rx2) - rx + bx):(rnd(mapdata->xs - 2) + 1);
+		*y = (ry >= 0) ? (rnd(ry2) - ry + by):(rnd(mapdata->ys - 2) + 1);
```

Comment:
Should be a new rnd_value [bx - rx, bx + rx] and a new rnd_value [1, mapdata->xs[
Same for y.